aten/src/ATen/core/Tensor.h

```diff
@@ -185,7 +185,7 @@ class CAFFE2_API Tensor {
   }
 
   at::MemoryFormat suggest_memory_format() const {
-    if (impl_->is_strides_like_channels_last()) {
+    if (!impl_->is_contiguous() && impl_->is_strides_like_channels_last()) {
```

Comment:
Well this would make NHWC the default format for everything, so it's too big of a change from today's behavior.

Reply:
Wait, no. This only suggest NHWC when it's not contiguous *and* it's striding like channels_last.

I'm actually defaulting to NCHW in ambiguous cases, so I'm making it more restrained.

Reply:
We are also testing that we are not accidentally making output channels last in the tests.
See
```python
self.assertTrue(ref_out.is_contiguous())
```

Reply:
Uh yeah I misread the conditional... Disregard this please.